# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

In [2]:
# TODO

from scipy import stats

print("H0= No existen diferencias significativas entre las dos dietas")
print("H1= Existen diferencias significativas entre las dos dietas")

diet1=[2.0,2.5,3.0,2.8,2.3,2.7,2.5]
diet2=[3.0,3.2,3.1,2.9,2.8,3.0,3.2]

t_value, p_value = stats.ttest_ind(diet1,diet2, alternative= "two-sided")

print(f"Con un valor p= {p_value:.3f} < 0.05 se rechaza la hipótesis nula (H0)")


H0= No existen diferencias significativas entre las dos dietas
H1= Existen diferencias significativas entre las dos dietas
Con un valor p= 0.004 < 0.05 se rechaza la hipótesis nula (H0)


## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

In [5]:
# TODO
Fertilizante_1 =[20, 21, 20,19,20]
Fertilizante_2 = [22,21,23,22,21]
Fertilizante_3 = [24,23,22,23,24]

f_value, p_value = stats.f_oneway(Fertilizante_1, Fertilizante_2, Fertilizante_3)

print (f"Con un valor p= {p_value:.4f} < 0.05 se rechaza la hipótesis nula (H0)")

print("H0= Las medias de todos los grupos son iguales ")
print("H1= Existen diferencias significativas entre las medias de los grupos")

#Una vez que se sabe que los fertilizantes son diferentes y puesto que los datos son respecto al rendimiento de parcelas identicas. Vamos a analizar cual es mejor que otro:

from scipy.stats import tukey_hsd

results= tukey_hsd(Fertilizante_1, Fertilizante_2, Fertilizante_3)
print(results)



Con un valor p= 0.0001 < 0.05 se rechaza la hipótesis nula (H0)
H0= Las medias de todos los grupos son iguales 
H1= Existen diferencias significativas entre las medias de los grupos
Tukey's HSD Pairwise Group Comparisons (95.0% Confidence Interval)
Comparison  Statistic  p-value  Lower CI  Upper CI
 (0 - 1)     -1.800     0.010    -3.143    -0.457
 (0 - 2)     -3.200     0.000    -4.543    -1.857
 (1 - 0)      1.800     0.010     0.457     3.143
 (1 - 2)     -1.400     0.041    -2.743    -0.057
 (2 - 0)      3.200     0.000     1.857     4.543
 (2 - 1)      1.400     0.041     0.057     2.743



El test ANOVA confirma diferencias significativas entre los 3 fertilizantes. 
El test de Tukey demuestra que:
El grupo 0 es el que tiene menor media: El valor statics siempre es negativo cuando se encuentra en primer lugar.
El grupo 2 es el que tiene una media de rendimiento superior a todos: El valor statics siempre e positivo cuando se encuentra en primer lugar.
El grupo dos es 3.2 unidades de media mayor que el grupo 0 y 1.4 unidades mayor que el grupo 1 

Todos los p values son inferiores a 0,05, por lo que todas las diferencias entre grupos son significativas.

Ninguno de los intervalos de confianza contienen al 0, por lo que demuestra diferencia significativa.

El fertilizante 3, que corresponde al grupo 2 es superior al resto.